pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -58,6 +58,45 @@ var (
 )
 
 // NewEndpointController returns a new *EndpointController.
+func NewEndpointControllerFromSharedInformer(podInformer framework.SharedInformer, client *clientset.Clientset) *EndpointController {
+	e := &EndpointController{
+		client: client,
+		queue:  workqueue.New(),
+	}
+
+	e.serviceStore.Store, e.serviceController = framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+				return e.client.Core().Services(api.NamespaceAll).List(options)
+			},
+			WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+				return e.client.Core().Services(api.NamespaceAll).Watch(options)
+			},
+		},
+		&api.Service{},
+		// TODO: Can we have much longer period here?
+		FullServiceResyncPeriod,
+		framework.ResourceEventHandlerFuncs{
+			AddFunc: e.enqueueService,
+			UpdateFunc: func(old, cur interface{}) {
+				e.enqueueService(cur)
+			},
+			DeleteFunc: e.enqueueService,
+		},
+	)
```

Comment:
> If we want to proceed with this PR, we should avoid this code duplication. But I guess this is just POC now, so never mind :)

My plan would be to switch over a couple controllers, prove that they work, then start phasing in the new approach and removing the old paths.  Probably once we have a standard way of tracking the shared informers for a given resource.